### PR TITLE
address code review comments for linux cluster templates

### PR DESCRIPTION
**Pull request by ShuaiYe2017**
Address the issues below in this PR
1. Directory for lambda functions
=>I cannot check in the zip file into "functions/packages/getVPCInfo/". Could you help me zip the file info_lambda.js?

2. Linux bastion template used should be a submodule of our Linux Bastion Quick Start (quickstart-linux-bastion). For how to add a submodule please see: https://aws-quickstart.github.io/design.html#submodules
The same should be done for the windows bastion (aka. remote desktop gateway) using the RD Gateway Quick Start (quickstart-microsoft-rdgateway).
=> done

3. Please name the templates as follows:
Windows New VPC: tableau-cluster-windows-master.template
Windows Existing VPC: tableau-cluster-windows.template
Linux New VPC: tableau-cluster-linux-master.template
Linux Existing VPC: tableau-cluster-linux.template
=> done

3. Call the VPC template from the submodules/quickstart-aws-vpc/ location and not from the templates/ directory.
The AMI mapping doesn't follow the changes we did on the single server one. Please name it the same and follow the alphabetic ordering and include the AMI element.
=> done

4. I see that you're combining the CentOS and Ubuntu deployment on these... why not do the same in the single server templates?
There is a trade off between combining and separating CentOS and Ubuntu cluster deployment scripts. Because cluster is huge, we decided to combining Centos and Ubuntu templates to avoid too much duplicates.
So our structure of cloudformation templates is:
single master => windows, ubuntu, centos
windows master => windows
linux master => centos, ubuntu.

5. Are you testing these templates before submitting the PRs?
I only did testing with the js file in the template. After getting info_lambda.js out of template, it's not easy for me to do the testing.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -2,3 +2,11 @@
 	path = submodules/quickstart-aws-vpc
 	url = git@github.com:aws-quickstart/quickstart-aws-vpc.git
 	branch = master
+[submodule "submodules/linux-bastion"]
```
Comment (santiagocardenas) [inline .gitmodules]: This must be "submodules/quickstart-linux-bastion"

File: .gitmodules [hunk 2]
```diff
@@ -2,3 +2,11 @@
...
+	path = submodules/linux-bastion
```
Comment (santiagocardenas) [inline .gitmodules]: This must be "submodules/quickstart-linux-bastion"

File: .gitmodules [hunk 3]
```diff
@@ -2,3 +2,11 @@
...
+	url = https://github.com/aws-quickstart/quickstart-linux-bastion.git
+	branch = master
+[submodule "submodules/windows-bastion"]
```
Comment (santiagocardenas) [inline .gitmodules]: This must be "submodules/quickstart-microsoft-rdgateway"

File: .gitmodules [hunk 4]
```diff
@@ -2,3 +2,11 @@
...
+	path = submodules/windows-bastion
```
Comment (santiagocardenas) [inline .gitmodules]: This must be "submodules/quickstart-microsoft-rdgateway"

File: templates/tableau-cluster-linux-master.template [hunk 5]
```diff
@@ -0,0 +1,706 @@
...
+            }
+        },
+        "WorkloadStack":{  
+            "Type":"AWS::CloudFormation::Stack",
+            "DependsOn":[  
+                "VPCStack"
+            ],
+            "Properties":{  
+                "TemplateURL":{  
+                    "Fn::Sub": "https://${QSS3BucketName}.s3.amazonaws.com/${QSS3KeyPrefix}templates/tableau_server_cluster_existing_vpc_linux.json"
```
Comment (santiagocardenas) [inline templates/tableau-cluster-linux-master.template]: This name is wrong...

File: templates/tableau-cluster-linux-master.template [hunk 6]
```diff
@@ -0,0 +1,706 @@
...
+            }
+        },
+        "BastionHost":{  
+            "Type":"AWS::CloudFormation::Stack",
+            "DependsOn":[  
+                "VPCStack"
+            ],
+            "Properties":{  
+                "TemplateURL":{  
+                    "Fn::Sub": "https://${QSS3BucketName}.s3.amazonaws.com/${QSS3KeyPrefix}templates/linux-bastion.template"
```
Comment (santiagocardenas) [inline templates/tableau-cluster-linux-master.template]: This should be calling from the submodule itself... the linux-bastion.template that's in there.

File: templates/tableau-cluster-linux-master.template [hunk 7]
```diff
@@ -0,0 +1,706 @@
...
+                            "Outputs.PublicSubnet1ID"
+                        ]
+                    },
+                    "PublicSubnet2ID":{  
+                        "Fn::GetAtt":[  
+                            "VPCStack",
+                            "Outputs.PublicSubnet2ID"
+                        ]
+                    },
+                    "VPCID":{
```
Comment (santiagocardenas) [inline templates/tableau-cluster-linux-master.template]: You'll want to include the QSS3BucketName and QSS3KeyPrefix parameters like in: 
https://github.com/aws-quickstart/quickstart-tableau-awsdata/blob/master/templates/tableau-awsdata-master.template#L958-L963

**Comment by santiagocardenas** [comment]
The Windows cluster master template needs updating now to call the standalone template in the Microsoft RD Gateway Quick Start. Take a look at this template which includes both a Linux and Windows bastion: https://github.com/aws-quickstart/quickstart-tableau-awsdata/blob/master/templates/tableau-awsdata-master.template

**Comment by santiagocardenas** [comment]
That's interesting that you werent able to create a lambda.zip file to include in your PR. I can help with it after merging the PR  if you are still having trouble.

File: templates/tableau-cluster-linux-master.template (hunk 6 above)
Comment (ShuaiYe2017) [reply templates/tableau-cluster-linux-master.template]: should it be "https://${QSS3BucketName}.s3.amazonaws.com/${QSS3KeyPrefix}submodules/quickstart-linux-bastion/templates/linux-bastion.template" ?

Comment (santiagocardenas) [reply templates/tableau-cluster-linux-master.template]: Correct. Plus ensure that the parameters are correct. Please see the example template I put in another comment for more info.

**Comment by santiagocardenas** [comment]
superseded by #13

**Pull request closed without merging**